### PR TITLE
Add filter for pending active task to redispatch

### PR DESCRIPTION
**Pull request by yux0**
**What changed?**
Add a task filter to retry tasks when the domain is pending active

**Why?**
During graceful failover, the cluster should not process task as active

**How did you test it?**

**Potential risks**
N/A

### CODE REVIEW

File: service/history/taskAllocator.go [hunk 1]
```diff
@@ -28,6 +28,7 @@ import (
 	"github.com/uber/cadence/common/log"
 	"github.com/uber/cadence/common/log/tag"
 	"github.com/uber/cadence/service/history/shard"
+	ctask "github.com/uber/cadence/service/history/task"
```
Comment (yycptt) [inline service/history/taskAllocator.go:31]: drop the import alias?

Comment (yux0) [reply service/history/taskAllocator.go:31]: the task is used as variable below

Comment (yycptt) [reply service/history/taskAllocator.go:31]: I see. I use `ctask` in other places for the `common/task` package as `c` stands for `common`.

Comment (yux0) [reply service/history/taskAllocator.go:31]: I see. Then I will try to find another alias

**Pull request merged**